### PR TITLE
Npv

### PR DESCRIPTION
**Pull request by mjburton11**
Ready for review.  @whoburg, I would like to know the following:

1) How is the overall content? Are there any big gaps that I'm not addressing? Do you also want a general example for N periods? 
2) Is there a specific latex format you would like me to use? 
3) Do you want me to try and make a gpkit model?

### CODE REVIEW

**Comment by bqpd** [comment]
Doesn't the program as written drive all `t_i` to 0?

**Comment by mjburton11** [comment]
Nope.   Here's a sample output.  Let me know what you think. 

```
Cost
----
 4.166e-08 [1/USD] 

Free Variables
--------------
NPV : 2.4e+07                           [USD]  net present value
 PV : 8.002e+06                         [USD]  present value    
  C : [ 1.04e+07  1.51e+07  4.28e+07 ]  [USD]  cash flow        
  t : [ 2.59      6.37      17.1     ]  [year] time             

Constants
---------
C_r : 4e+06  [USD/year] cash flow rate
  r : 0.1    [1/year]   interest rate 

Sensitivities
-------------
  r : 1  interest rate 
C_r : -1 cash flow rate
```

**Comment by bqpd** [comment]
Interesting. Okay. Should that be `N*PV` in the first constraint?

**Comment by bqpd** [comment]
This is a nifty program to look at for following pressures!

File: npv/npv.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        N = 20 # number of payments
+        NPV = Variable('NPV', 'USD', 'net present value')
+        PV = Variable('PV', 'USD', 'present value')
+        C = VectorVariable(N, 'C', 'USD', 'cash flow')
+        C_r = Variable('C_r', 4e6, 'USD/years', 'cash flow rate')
+        r = Variable('r', np.linspace(0.01,0.1,10), '1/years', 'interest rate')
+        t = VectorVariable(N, 't', 'years', 'time')
+       
+        constraints = [3*PV >= NPV,
```
Comment (mjburton11) [inline npv/npv.py]: @bqpd this should definitely be N, thanks for catching that.

File: npv/npv.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# inPDF: skip
+
+from numpy import pi
+from gpkit import VectorVariable, Variable, Model, units
+import gpkit
+
+class NPV(Model):
+
+
+```
```
Comment (whoburg) [inline npv/npv.md]: this md file doesn't look like it's relevant -- should it be removed?

File: npv/npv.tex [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+\text{C}$_r$: [\$/\text{time}] \text{ Cash flow rate} \\
+\text{PV}: [\$] \text{ Value of an investment at the present time} \\
+t : [\text{time}] \text{ Time}
+\end{tabbing}
+
+\section*{Problem Set Up}
+
+Let's assume that we want the Net Present Value (NPV) to be \$10 Million.  NPV can be expressed by:
+
+\[ \text{NPV} = \displaystyle\sum\limits_{i=0}^N \text{PV}_i \] .
```
Comment (whoburg) [inline npv/npv.tex]: period should go inside the math - it's landing on a new line currently.

File: npv/npv.tex [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ \end{array} \]
+ 
+ \item Objective 
+ 
+ \[ \text{maximize}: \text{NPV} \]
+ 
+ \item Constraints
+ \[ \begin{array}{ll}
+ & NPV \leq 3PV \\
+ & C \geq PV\left(1 + rt_i + \frac{rt_i^2}{2!} + \frac{rt_i^3}{3!} \right) \\
```
Comment (whoburg) [inline npv/npv.tex]: + ...  (need ..., don't cut off the taylor expansion)

**Comment by whoburg** [comment]
@mjburton11, looking good. added a few comments.

This would be a good use case for the markdown parsing.

Would also be good to produce some plots showing how the integrated cash flow over each period adds up to the npv.

**Comment by bqpd** [comment]
@mjburton11, let's meet up sometime and try doing this in markdown!

**Comment by mjburton11** [comment]
@bqpd sounds great!

**Comment by whoburg** [comment]
I see the new md file -- should the old tex and python files get removed now?

**Pull request merged**